src/App/Fossa/VSI/DynLinked/Internal/Lookup/APK.hs

```diff
@@ -0,0 +1,163 @@
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TupleSections #-}
+
+module App.Fossa.VSI.DynLinked.Internal.Lookup.APK (
+  lookupDependencies,
+  constructLookupTables,
+  tryLookup,
+  SyftData (..),
+  SyftArtifact (..),
+  SyftArtifactMetadata (..),
+  SyftArtifactMetadataFile (..),
+  SyftLookupTable (..),
+) where
+
+import App.Fossa.VSI.DynLinked.Types (DynamicDependency (..), LinuxPackageManager (..), LinuxPackageMetadata (..), ResolvedLinuxPackage (..))
+import App.Fossa.VSI.DynLinked.Util (runningLinux)
+import Control.Algebra (Has)
+import Control.Effect.Diagnostics (Diagnostics, fatalText)
+import Control.Monad (join)
+import Control.Monad.Trans.Maybe (MaybeT (MaybeT), runMaybeT)
+import Data.Aeson (FromJSON, Result (Error, Success), ToJSON, Value, fromJSON, object, parseJSON, toJSON, withObject, (.:), (.=))
+import Data.Either (partitionEithers)
+import Data.Map (Map)
+import Data.Map qualified as Map
+import Data.String.Conversion (toText)
+import Data.Text (Text)
+import Effect.Exec (AllowErr (Never), Command (..), Exec, execJson)
+import Path (Abs, Dir, File, Path)
+
+-- | The idea here is that we look up what paths we can with apt and turn them into @DynamicDependency@.
+-- We then hand back leftovers and lookup results for the next resolution function.
+lookupDependencies :: (Has Diagnostics sig m, Has Exec sig m) => Path Abs Dir -> [Path Abs File] -> m ([Path Abs File], [DynamicDependency])
+lookupDependencies _ files | not runningLinux = pure (files, [])
+lookupDependencies root files = do
+  syft <- runSyft root
+  table <- constructLookupTables syft
+  partitionEithers <$> traverse (tryLookup table) files
+
+tryLookup :: (Has Diagnostics sig m) => SyftLookupTable -> Path Abs File -> m (Either (Path Abs File) DynamicDependency)
+tryLookup SyftLookupTable{..} file = fmap (maybeToRight file) . runMaybeT $ do
```

Comment:
You can probably inline the definition of `maybeToRight`, if you use `maybe` instead of pattern matching.
```suggestion
tryLookup SyftLookupTable{..} file = fmap (maybe (Left file) Right) . runMaybeT $ do
```

Reply:
supersesed by this comment: https://github.com/fossas/fossa-cli/pull/786/files/fc6986fc10d6fa0f29e0e58503875bbea36e2203#r798123119